### PR TITLE
[clang-cpp-14] Support variable template specialization. Fixes #1728

### LINKED ISSUE #1728: [cpp] Support `VarTemplate` and `VarTemplateSpecialization`
## VarTemplate

cpptest.pre.cpp:
```cpp
template <typename> int a;
```

Running

```
esbmc --cppstd 17 --input-file cpptest.pre.cpp --no-abstracted-cpp-includes --no-library
```

Fails with:

```
ERROR: unrecognized / unimplemented clang declaration VarTemplate
```

## VarTemplateSpecialization

After patching esbmc to ignore `VarTemplate` it fails with another problem:

cpptest.pre.cpp:
```cpp
template <typename> int a;

int b = !a<int>;
```

Running

```
esbmc --cppstd 17 --input-file cpptest.pre.cpp --no-abstracted-cpp-includes --no-library
```

Fails with:

```
ERROR: unrecognized / unimplemented clang declaration VarTemplateSpecialization
```

### PR DESCRIPTION
**Pull request by intrigus-lgtm**
Implements https://github.com/esbmc/esbmc/issues/1728#issuecomment-1994371059 and adds tests to the new esbmc-cpp14 directory.

### CODE REVIEW

**Review by fbrausse — APPROVED** [review]
LGTM, many thanks for submitting this PR to implement support for the variable templates.

File: regression/CMakeLists.txt [hunk 1]
```diff
@@ -114,6 +114,9 @@ set(REGRESSIONS_CPP11 esbmc-cpp11/cpp
                       esbmc-cpp11/template
                       )
 
+# list of C++14 test suites
+set(REGRESSIONS_CPP11 esbmc-cpp14/template)
```
Comment (fbrausse) [inline regression/CMakeLists.txt]: Sorry, I missed this earlier: This needs to either include "${REGRESSIONS_CPP11}" or be a new variable that's being used below.

**Review by fbrausse — CHANGES_REQUESTED** [review]

File: regression/CMakeLists.txt (hunk 1 above)
Comment (intrigus-lgtm) [reply regression/CMakeLists.txt]: Very good catch, copy and paste artifact...

Comment (intrigus-lgtm) [reply regression/CMakeLists.txt]: Fixed.

**Review by lucasccordeiro — APPROVED** [review]
Thanks for submitting this PR, @intrigus-lgtm.

**Pull request merged**